code.py

```diff
@@ -4,24 +4,67 @@
 It enables you to more easily create complex layouts and has other advanced 
 features, including double-tap support.
 """
+try:
+    from apps_private.default import DEFAULT_APP, SELECT_APPS
+except Exception:
+    from apps.home import HomeApp
 
-from apps.home import HomeApp
-from app_pad import AppPad
-from commands import AppSwitchException
-from constants import OS_MAC, OS_SETTING, PREVIOUS_APP_SETTING
+    DEFAULT_APP = HomeApp
+    SELECT_APPS = {}
```

Comment:
I really like this `private_apps` idea. Your `DEFAULT_APP` and `SELECT_APPS` constants make me think of a Django settings module.

What if we created a versioned `default_settings.py` file containing the default configuration here:
```py
from apps.home import HomeApp

DEFAULT_APP = HomeApp
SELECT_APPS = {}
```
Then we could have an ignored `settings.py` file to contain what is in `apps_private.default`. 

Then here we would do something like 
```
try:
    from settings import DEFAULT_APP, SELECT_APPS
except Exception:
    from default_settings import DEFAULT_APP, SELECT_APPS
```

Reply:
This is what I was trying to do with the config file, but I like the idea of following a django-like `settings.py` in place of that. 

Here are some of the ideas I had:
1. Private apps and their related settings are kept in a single .gitignored folder. The framework defines an interface/entry point but is otherwise agnostic to its contents. This would allow users to pull updates to the project without impacting their Apps. Potentially the private_apps dir could be its own git project maintained by the user. 
2. Settings are configured with defaults in a single place (`settings.py`). 
    1. As much as possible, settings should be configured there so that less technical users could make changes they may want, and they can be easily configured as described: 
    2. Settings can be overridden hierarchically. `/settings.py` is overridden by `private_apps/settings.py` is overridden by data passed over the serial data connection. I am basically following the pattern laid out in 3 scoops of Django.
    3. Maybe we can move the settings that are used in each app to `settings.py` and make the framework reactive ala vue/vuex. Instead of a switch app command, updating the `settings.current_app` reacts as you would expect, redrawing the screen etc. One thing at a time though 😄

Reply:
OK, cool. I like where you're headed with the settings. I'll put together a PR with some basic functionality there. I'm hesitant to put too much in the settings, though; I'm worried that it might limit what is possible with the apps if the settings are held externally.

I'm also not terribly concerned with supporting less technical users. I feel like those users will use the original adafruit code. With that being said, I'm all for making the code more intuitive. So hopefully that will make things easier for everyone.

Reply:
https://github.com/kbaskett248/adafruit_macropad/pull/10

This is a pretty simple PR that enables user configurations. I went with a `user` folder over `private_apps` because I liked it a bit more.

I spent some time thinking about moving `current_app` into the `settings`. I couldn't make it work in a way I liked. The main problem I have is that we need to raise an exception when the setting changes. But, if we raise that exception before we're in the run loop, everything breaks. If you have any ideas there, I'm open to it.

Reply:
If the single first app is the entry point, why move current_app out of code.py at all?

Reply:
Just going off your comment about moving the `current_app` into settings.